multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/IncrementalAppInstancesUpdateStep.java

```diff
@@ -0,0 +1,149 @@
+                                                            .getInstances();
+        var incrementalAppInstanceUpdateConfigurationBuilder = ImmutableIncrementalAppInstanceUpdateConfiguration.builder()
+                                                                                                                 .newApplication(application)
+                                                                                                                 .newApplicationInstanceCount(idleApplicationInstances.size());
+        if (oldApplication == null) {
+            context.getStepLogger()
+                   .info(Messages.DUE_TO_MISSING_PRODUCTIVE_DEPLOYED_APPLICATION_OF_MODULE_0_THE_NEW_APPLICATION_WILL_BE_SCALED_IN_STANDARD_WAY,
+                         application.getModuleName());
+            client.updateApplicationInstances(application.getName(), application.getInstances());
+            incrementalAppInstanceUpdateConfigurationBuilder = ImmutableIncrementalAppInstanceUpdateConfiguration.builder()
+                                                                                                                 .newApplication(application)
+                                                                                                                 .newApplicationInstanceCount(application.getInstances());
+            context.setVariable(Variables.INCREMENTAL_APP_INSTANCE_UPDATE_CONFIGURATION,
+                                incrementalAppInstanceUpdateConfigurationBuilder.build());
+            context.setVariable(Variables.ASYNC_STEP_EXECUTION_INDEX, 1);
+            return StepPhase.POLL;
+        }
+        int oldApplicationInstanceCount = client.getApplicationInstances(oldApplication)
+                                                .getInstances()
+                                                .size();
+        incrementalAppInstanceUpdateConfigurationBuilder.oldApplication(oldApplication)
+                                                        .oldApplicationInitialInstanceCount(oldApplicationInstanceCount)
+                                                        .oldApplicationInstanceCount(oldApplicationInstanceCount);
+        context.setVariable(Variables.INCREMENTAL_APP_INSTANCE_UPDATE_CONFIGURATION,
+                            incrementalAppInstanceUpdateConfigurationBuilder.build());
+        if (idleApplicationInstances.size() >= application.getInstances()) {
+            context.getStepLogger()
+                   .info(Messages.APPLICATION_0_ALREADY_SCALED_TO_THE_DESIRED_1_INSTANCES, application.getName(),
+                         application.getInstances());
+            return StepPhase.DONE;
+        }
+        return StepPhase.POLL;
+    }
+
+    private void checkWhetherLiveAppNeedsPolling(ProcessContext context, CloudControllerClient client, CloudApplication cloudApplication) {
+        context.setVariable(Variables.ASYNC_STEP_EXECUTION_INDEX, 2);
```

Comment:
Extracted the in methods to avoid comments. It is not always required to wait the old or the new app. During the first execution of the incremental update the new app is running on one instance, so we do not need to check the instances